packages/js/product-editor/src/components/iframe-editor/keyboard-shortcuts/register-keyboard-shortcuts.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as keyboardShortcutsStore } from '@wordpress/keyboard-shortcuts';
+// @ts-expect-error missing types.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { isAppleOS } from '@wordpress/keycodes';
+
+export const RegisterKeyboardShortcuts = () => {
```

Comment:
Can we transform this into a hook? Since it's not rendering anything and is only executing a side effect, it would make more sense to handle this code as a hook rather than as a normal component.

Reply:
I thought the same thing. But, this needs to be under the `<BlockEditorProvider>`, so that it is in the right context. I could create a wrapper component that used the hook under there, but that wrapper component would also render nothing other than its children.

This is how Gutenberg handles this in its editors.

Reply:
I usually prefer using a HOC over a component that doesn't render anything (functional component), but given that we use that pattern sporadically, it's good as it is.